src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -8716,8 +8716,8 @@ private static void AssertTupleTypeEquality(TypeSymbol tuple)
                     if (i != j)
                     {
                         Assert.NotSame(members[i], members[j]);
-                        Assert.False(members[i].Equals(members[j]));
-                        Assert.False(members[j].Equals(members[i]));
+                        Assert.True(!members[i].Equals(members[j]) || members[i].Name != members[j].Name);
+                        Assert.True(!members[j].Equals(members[i]) || members[j].Name != members[i].Name);
```

Comment:
Equality on `TupleErrorFieldSymbol` doesn't care about the element name (which seems reasonable, but I'm open to discuss). After this change, it is possible to create two `TupleErrorFieldSymbol`s with same tuple element index, which are therefore equal.

Here's the relevant code in `TupleTypeSymbol.CreateMembers`:
```C#
     if (name != defaultName)
                    {
                        members.Add(new TupleErrorFieldSymbol(this, defaultName, i, null, _elementTypes[i], diagnosticInfo));
                    }

                    members.Add(new TupleErrorFieldSymbol(this, name, i,
                                                      _elementLocations.IsDefault ? null : _elementLocations[i],
                                                      _elementTypes[i],
                                                      diagnosticInfo));
```

Those two used to use different element indices, but now it is fixed so they both use the proper index.